luaui/Widgets_BAR/cmd_select_commander.lua

```diff
@@ -0,0 +1,28 @@
+function widget:GetInfo()
+  return {
+    name = "SelectCommander",
+    desc = "Press ctrl+c to select the commander",
+    author = "Teifion",
+    date = "26/10/2020",
+    license = "GNU GPL, v2 or later",
+    layer = 0,
+    enabled = true
+  }
+end
+
+function widget:Update()
+  local t = Spring.GetGameSeconds()
+  if Spring.GetSpectatingState() then
+    widgetHandler:RemoveWidget(self)
+    return
+  end
+  if (t > 0) then
+    local unitArray = Spring.GetTeamUnits(Spring.GetMyTeamID())
+    for key, unit in pairs(unitArray) do
```

Comment:
Aren't the keys actually just integers? If so the pairs could be optimized to just be a regular array iteration
```suggestion
    for i = 1, #unitArray do
        local unit = unitArray[i]
```